includes/class-micropub-endpoint.php

```diff
@@ -249,12 +269,12 @@ private static function query_handler( $user_id ) {
 					}
 					break;
 				default:
-					static::error( 400, 'unknown query ' . static::$input['q'] );
+					return new WP_Micropub_Error( 'invalid_request', 'unknown query', 400, static::$input );
 			}
 		}
 
 		do_action( 'after_micropub', static::$input, null );
-		static::respond( 200, $resp );
+		return rest_ensure_response( $resp );
```

Comment:
interesting. is this useful? if so, should we do it at the end of `post_handler` too?